### DISCUSSION TITLE
Removeparam not working

### DISCUSSION THREAD

**Discussion by Vanerio** [uBlockOrigin/uBlock-issues#2559] [General]
Hi!
First of all, I am not an expert.
I am trying to remove 's' parameter in this link:
<https://twitter.com/Twitter/status/1623411536243965954?s=20>

The thing is that in one of my Brave profiles it works, the URL gets modified. But in other profile, it does not work.
I have tried to export-import uBlock configuration file, so it has the exact same settings, but it does not work either.
I have tried to replicate all Brave conf and it does not work either.

One thing I could notice in uBlock´s Logger is the following: The profile that removes 's' param has the following requests:
```
| +0 |  |  | twitter.com | 1 | get | doc | `https://twitter.com/Twitter/status/1623411536243965954` |
| +0 | `https://twitter.com/Twitter/status/1623411536243965954` | << | twitter.com | 1 | get | doc | `https://twitter.com/Twitter/status/1623411536243965954?s=20` |
| +0 | twitter.com/*?$removeparam=s | -- | twitter.com | 1 | get | doc | `https://twitter.com/Twitter/status/1623411536243965954?s=20` |
```

And the profile that does not remove 's' param has the following requests:
```
+0	https://twitter.com/Twitter/status/1623411536243965954?prefetchTimestamp=1679734921862	<<	twitter.com	0,1	get	xhr	https://twitter.com/Twitter/status/1623411536243965954?s=20&prefetchTimestamp=1679734921862
+0	twitter.com/*?$removeparam=s	--	twitter.com	0,1	get	xhr	https://twitter.com/Twitter/status/1623411536243965954?s=20&prefetchTimestamp=1679734921862
```

So in this last one i can see that:
- A prefetchTimestamp has been added to the url.
- There is not doc request, it is named xhr instead.
- s parameter is not removed.

I don´t really know how to make it work. 
Could someone please help me with that?
Thank you very much!

**uBlock version: 1.48.0**

**Comment by ghost** [comment]
Can you confirm it's not removed by checking in browser dev tools or console? 

I think it's just that the second case with xhr is requested by service worker and address in url bar is modified (or not) by java script, so the parameter actually is removed from network request (as logged by `<<` filter in the logger), but you just not see it in address bar.

**Reply by Vanerio** [reply]
This is what I have checked:
Brave DevTools -> Network:
There it appears: https://twitter.com/Twitter/status/1623411536243965954?s=20&prefetchTimestamp=1679737801243
And in  its Payload: 
```
Query string parameters:
s:20
prefetchTimestamp: 1679737801243
```

If I am checking it wrong, please let me know!
Thank you gwarser!

**Reply by ghost** [reply]
This can be prefetch request, so it will not be blocked - Chromium issue. Someone will need to repro and check.

**Reply by Vanerio** [reply]
But if it is a Chromium issue why is it working fine in my other Brave profile?  😵
Edit: I mean I am doing the exact same thing in both profiles in the same browser.

**Reply by gorhill** [reply]
I am not able to reproduce any issue. Appearing in the dev tools does not mean it's not working if it is followed by an internal redirect request as a result of uBO redirecting it. Anyway, as said I am not able to reproduce any issue, either the steps to reproduce are incomplete or there are missing details about what is happening (and also we always ask about troubleshooting information).

**Reply by Vanerio** [reply]
I am not able to reproduce it either in the working profile (to make it fail) even after importing uBlock settings.
As I said, I am not an expert at all. Please, let me know the exact steps I must follow to provide you the information needed.

**Reply by gothic-bum** [reply]
Post the troubleshooting information:  
Click the [uBO icon](https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface)  > Click the [Chat icon](https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface#the-tools) > Click "Troubleshooting Information"

**Reply by Vanerio** [reply]
Ok, here it is! But first let me tell you that, even if I disable uBlock and I fresh install 'uBlock Origin development build' and create just the removeparam filter (everything else is default settings), it does not work either!
So it may be a Brave configuration the one avoiding this filter to be applied. But since I have already tried to replicate the configuration in the other profile, I don't really know what it is or what to do.
I would also like to say that removeparam is applied as expected in other websites.
Thank you!


```
uBlock Origin: 1.48.0
Chromium: 111
filterset (summary): 
  network: 211570
  cosmetic: 119567
  scriptlet: 20016
  html: 0
listset (total-discarded, last-updated): 
  added: 
    https://filters.adtidy.org/extension/ublock/filters/3.txt: 39329-3448, 21h.11m
    https://raw.github.com/reek/anti-adblock-killer/master/anti-adblock-killer-filters.txt: 0-0, 21h.11m
    https://raw.githubusercontent.com/DandelionSprout/adfilt/master/LegitimateURLShortener.txt: 2143-1, 21h.11m
    https://raw.githubusercontent.com/Spam404/lists/master/adblock-list.txt: 7250-0, 21h.11m
    https://s3.amazonaws.com/lists.disconnect.me/simple_malvertising.txt: 2735-45, 21h.11m
    https://winhelp2002.mvps.org/hosts.txt: 8729-578, 21h.11m
    https://www.fanboy.co.nz/enhancedstats.txt: 2990-22, 21h.11m
    adguard-spyware-url: 954-134, 2d.23h.39m
    curben-phishing: 60722-1, 21h.17m
    dpollock-0: 11342-1742, 22m
    fanboy-annoyance: 82138-393, 4d.10h.31m
    fanboy-cookiemonster: 46232-46232, 4d.10h.32m
    fanboy-social: 22248-22248, 3d.9h.15m
    fanboy-thirdparty_social: 68-19, 4d.10h.33m
  default: 
    user-filters: 503-0, never
    easylist: 64466-581, 1d.15h.20m
    easyprivacy: 30957-12306, 3d.23h.42m
    plowe-0: 3755-2544, 7d.14h.1m
    spa-0: 3550-14, 21h.19m
    spa-1: 5133-205, 2d.10h.49m
    ublock-abuse: 78-0, 10d.12h.21m
    ublock-badware: 5081-144, 2d.10h.48m
    ublock-filters: 31999-180, 1d.15h.19m
    ublock-privacy: 292-65, 10d.12h.20m
    ublock-quick-fixes: 434-7, 21h.16m
    ublock-unbreak: 1938-71, 1d.15h.18m
    urlhaus-1: 7248-3, 21h.15m
filterset (user): [array of 493 redacted]
switchRuleset: 
  added: [array of 146 redacted]
hostRuleset: 
  added: [array of 265 redacted]
modifiedUserSettings: 
  advancedUserEnabled: true
  cloudStorageEnabled: true
  showIconBadge: false
modifiedHiddenSettings: [none]
supportStats: 
  allReadyAfter: 427 ms (selfie)
  maxAssetCacheWait: 202 ms
```

**Reply by gorhill** [reply]
We need exact steps to reproduce, as pointed out, I can't reproduce any issue. Nothing will happen until you find a way for us to reproduce.

**Reply by Vanerio** [reply]
Do you know what browser settings might interfere with this?

**Reply by Vanerio** [reply]
I still don´t know how to reproduce it.
But I found out that removing CacheStorage folder allows removeparam to work, but only 1 time (until content is created again):
`C:\Users\{username}\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default\Service Worker\CacheStorage`

Any help on what browser settings may be affecting will be welcome.
Thank you.

**Reply by krystian3w** [reply]
Check if the service worker on Twitter has a stable name, e.g. `/sw.js` / `/serviceWorker.js` - you can temporarily block it to see if it will make the parameter removal effective.

**Reply by Vanerio** [reply]
Not really sure how to do that.
In `DevTools/Application/Service Workers` there is only a file named `1623411536243965954`

Edit: OK I checked the same thing in the profile that works and it says `sw.js`, but in the profile NOT working the file has this random number name.

**Reply by krystian3w** [reply]
try save in uBo panel ("My Filters"):

```adb
/sw.js$domain=twitter.com|/^twitter[\w].onion/,_,important
```

Close Brave, remove directory manually, open uBo with empty tab like `about:blank`, waiting as uBo icon again will be "brownish/redish" and opens twitter.
If it crashes again on the second attempt, maybe the service worker is still sitting in another file.

**Reply by ghost** [reply]
Brave implementation of Tor?

**Reply by Vanerio** [reply]
Done! But the same keeps happening

**Reply by Vanerio** [reply]
> Brave implementation of Tor?

I am not using Tor at all (not Brave incognito)

**Reply by krystian3w** [reply]
I added the onion address for an example as if they were serving an identical code.

**Reply by ghost** [reply]
And it works fine...

**Reply by Vanerio** [reply]
I tried adding the same exact filter to Brave Shields (`twitter.com/*?$removeparam=s`), and it works perfectly fine...so I am really lost at what could be wrong.

**Reply by iam-py-test** [reply]
First, I _think_ removing the `s` parameter from Twitter causes problems. Second (and maybe I missed something reading through this thread, if so, sorry), but this _could_ be due to Twitter's use of history.replaceState which bypasses uBo's filtering.

**Reply by krystian3w** [reply]
Maybe level of "abstraction", maybe gwarser rewrite this filter into DNR rule to test in modified uBo Lite?

As for me, you can try to force the shields to work only on twitter or see if deselecting all lists deactivates the protection that quarrels with uBo MV2.

**Reply by Vanerio** [reply]
Removing Brave Shields in Twitter does not help to uBo.

To iam-py-test: About your second point, as I said, the filter should work for you, and certanly it works for me in other profiles. So I guess the conflict comes from some Brave configuration and not an uBo configuration or Twitter.

Thank you all of you for trying to help!

**Reply by krystian3w** [reply]
You are not supposed to disable them completely, but limit them so that only one `removeparam` works.

To rule out whether the shield system or something specific to Brave is interfering, you can use Vivaldi or Iron Browser (e.g. https://portableapps.com/apps/internet/iron_portable).

**Reply by Vanerio** [reply]
It does not help either. Anyway, I have the exact same Brave Shield settings in my other Brave profile.
BUT I've just noticed something: the filter doesn't work when I'm logged in to Twitter. As soon as I log out, the filter works (I mean uBo filter, not Brave)!. So maybe it has something to do with uBo and Twitter?
